nupic/algorithms/anomaly.py

```diff
@@ -140,6 +148,17 @@ def compute(self, activeColumns, predictedColumns,
     if self._movingAverage is not None:
       score = self._movingAverage.next(score)
 
+    # apply binary discretization if required
+    if self._binaryThreshold is not None:
+      if score >= self._binaryThreshold:
+        score = 1.0
+      else:
+        score = 0.0
+
+    # final check on score within [0,1]
+    score = int(score * 100000000)/float(100000000) # 7 dec places
```

Comment:
sorry, my bad. but the problem was sum(anomalyScore) was negative... should I remove the rounding and let assert prove/catch it?